components/ca-import.js

```diff
@@ -0,0 +1,727 @@
+            // ensure its always csv or xlsx
+            // eslint-disable-next-line no-param-reassign
+            value = (value !== 'xlsx') ? value = 'csv' : 'xlsx';
+
+            this.setAttribute('type', value);
+            this.input.setAttribute('accept', `.${value}`);
+            this.link.setAttribute('download', `template.${value}`);
+        }
+
+        /**
+         * Gets convert url
+         * @param {string} convert url
+         */
+        get convertUrl() {
+            return this.getAttribute('convert-url') || '/ca-import';
+        }
+
+        /**
+         * Sets convertUrl
+         * @param {string} value, defaults to ca-import
+         */
+        set convertUrl(value = '/ca-import') {
+            this.setAttribute('convert-url', value);
+            this.form.action = value;
+        }
+
+        /**
+         * Gets end point that creates excel file from headers - only required for type=xls
+         * @returns {string} url
+         */
+        get templateConversionUrl() {
+            return this.getAttribute('template-conversion-url') || '/excel-download';
+        }
+
+        /**
+         * Sets end point that creates excel file from headers - only required for type=xls
+         * @param {string} value, url value
+         */
+        set templateConversionUrl(value = '/excel-download') {
+            this.setAttribute('template-conversion-url', value);
+            this.form.action = value;
+        }
+
+
+        /**
+         * Called when saving, makes an Ajax post.
+         * @returns {undefined} nothing.
+         */
+        save() {
+            if (this.saveUrl) {
+                componentSupport
+                    .request({
+                        url: this.saveUrl,
+                        type: 'POST',
+                        dataType: 'json',
+                        data: this.data,
+                        contentType: 'application/json'
+                    })
+                    .then(data => {
+                        trace('CSV file converted and posted');
+                        this.onsave.call(this, data);
+                    });
+            }
+        }
+
+       /**
+        * Executes once data has been posted to the server
+        * @param {array} data, table data
+        * @returns {undefined} nothing.
+        */
+        onsave(data) {
+            trace('saved');
+        }
+
+       /**
+        * Executes when the user adds a file
+        * @returns {undefined} nothing.
+        */
+        onfileadded() {
+            trace('user added file containing data');
+        }
+
+       /**
+        * Renders the HTML view of the data if present
+        * @param {array} headers to use for table
+        * @param {array} data to use for rows
+        * @returns {undefined} nothing. Builds table to DOM
+        */
+        render(headers, data) {
+            // eslint-disable-next-line no-param-reassign
+            headers = headers || this.headers;
```

Comment:
Think we can make this a default?
render(headers = this.headers, data = this._data)